src/현정빈/Myclass.java

```diff
@@ -0,0 +1,148 @@
+package 현정빈;
+
+import jun.ManagementSystem;
+import jun.Member;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class Myclass implements ManagementSystem {
+    HashMap<Integer,Member> map = new HashMap<>(); //컬렉션 중 Map을 선택하여 HashMap 사용
+    // 키는 integer 타입, 값은 Member 타입인 키-값쌍 구조의 map 생성
+
+    /**
+     * 저장소에 멤버를 추가하는 메서드,
+     * 이미 존재하는 id라면 추가에 실패해야 함,
+     * 추가가 성공하면 true를 반환
+     */
+    @Override
+    public boolean addMember(String name, int age, int id) {
+        Member newMember = new Member(name, age, id);
+        if(!map.containsKey(id)) {//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 false라면
+            map.put(id, newMember); //map에 지정된 키(id)와 값(객체 newMember)을 저장
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * 저장소에서 멤버를 삭제하는 메서드,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean removeMember(int id) {
+        if(!map.containsKey(id)){//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 false라면
+            return false;
+        }
+        else {
+            map.remove(id); //지정된 id와 연결된 값을 map에서 제거
+            return true;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 1,
+     * id를 기준으로 멤버를 찾아서, 이름을 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, String name) {
+        if(map.containsKey(id)) {//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setName(name);//map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 2,
+     * id를 기준으로 멤버를 찾아서 나이를 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, int age) {
+        if(map.containsKey(id)) {//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setAge(age);//map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 3,
+     * id를 기준으로 멤버를 찾아서 이름과 나이를 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, String name, int age) {
+        if(map.containsKey(id)) { //지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setName(name); //map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            map.get(id).setAge(age);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * id를 통해 멤버를 검색하는 메서드
+     */
+    @Override
+    public Member findById(int id) {
+        if(map.containsKey(id)){ //맵이 지정된 id를 포함하고 있는지 여부 반환, 반환된 값이 true라면
+            return map.get(id); //지정된 id에 연결된 값을 반환
+        }
+        return null;
+    }
```

Comment:
```suggestion
    @Override
    public Member findById(int id) {
        return map.get(id); // 지정된 id에 연결된 값을 반환, 없으면 null 반환
}
```
`get` 메서드는 `id`가 없으면 자동으로 `null`을 반환하기 때문에 이렇게 표현해도 충분 할 것 같습니다!